core/src/test/java/org/veriblock/sdk/ServiceSecurityPayloadTests.java

```diff
@@ -103,26 +103,35 @@ public void normalAddPayloadTest() throws BlockStoreException, SQLException {
         }
 
         ///HACK: we skip block difficulty validation. Will be fixed in future.
-        security.getBitcoinBlockchain().setSkipValidateBlocksDifficulty(false);
+        security.getBitcoinBlockchain().setSkipValidateBlocksDifficulty(true);
         security.getVeriBlockBlockchain().setSkipValidateBlocksDifficulty(true);
 
         // assume no error is thrown
         security.addPayloads(blockIndex, vtbPublications, altPublications);
-        
-        ///TODO: removePayloads throws an exception. Fix.
-        /*
+
         security.removePayloads(blockIndex);
         
-        // now we can see that only BTC validation throws an error
+        // check that the BTC difficulty validation throws an error
+        security.getBitcoinBlockchain().setSkipValidateBlocksDifficulty(false);
+        security.getVeriBlockBlockchain().setSkipValidateBlocksDifficulty(true);
+
+        try {
+            security.addPayloads(blockIndex, vtbPublications, altPublications);
```

Comment:
why second "addPayloads" shoud fail?

Reply:
calculators don't handle/aren't configured to handle the contexts